storage/range.go

```diff
@@ -838,46 +864,46 @@ func (r *Range) applyRaftCommand(index uint64, originNodeID multiraft.NodeID, ar
 	return reply.Header().GoError()
 }
 
-// startGossip periodically gossips the cluster ID if it's the first
-// range. This function should only be called if this range replica is
-// the raft leader.
-func (r *Range) startGossip(stopper *util.Stopper) {
-	stopper.RunWorker(func() {
-		r.maybeGossipClusterID()
-		r.maybeGossipFirstRange()
-
-		ticker := time.NewTicker(ttlClusterIDGossip / 2)
-		for {
-			select {
-			case <-ticker.C:
-				if held, expired := r.HasLeaderLease(r.rm.Clock().Now()); held && !expired {
-					r.maybeGossipClusterID()
-					r.maybeGossipFirstRange()
-				}
-			case <-stopper.ShouldStop():
-				return
-			}
-		}
-	})
-}
+// maybeGossipFirstRangeWithLease is periodically called and gossips the
+// sentinel and first range metadata if the range has the lease and is the
+// first range. Since one replica in the first range should always gossip this
+// information, a lease is acquired if there is no active lease.
+func (r *Range) maybeGossipFirstRangeWithLease() {
+	// If no Gossip available (some tests) or not first range: nothing to do.
+	if r.rm.Gossip() == nil || !r.isInitialized() || !r.IsFirstRange() {
+		return
+	}
+	timestamp := r.rm.Clock().Now()
+	held, shouldGetLease := r.HasLeaderLease(timestamp)
```

Comment:
Renaming this shouldGetLease doesn't make the code clearer. Just keep calling it `expired` I think would be better.

This logic is indistinguishable from redirectOnOrAcquireLeaderLease. Why not use that instead of duplicating the logic here? You could just change the parameters that method accepts to

    func (r *Range) redirectOnOrAcquireLeaderLease(timestamp proto.Timestamp, reason string) error

And supply `"on read or write pressure"` in one case and `"for first range gossip"` for `reason`